### PR TITLE
add option to disable security + rename of Enable* annotations

### PR DESCRIPTION
**Pull request by domivds**

### CODE REVIEW

**Review by christinaspachiou — APPROVED** [review]

File: security/src/main/java/zone/cogni/lib/security/common/SecurityConfiguration.java [hunk 1]
```diff
@@ -30,16 +31,22 @@ private void init() {
...
 
   @Configuration
   @ConditionalOnProperty(name = "cognizone.security.auth-method", havingValue = "saml2")
-  @EnableSaml2
+  @EnableSecuritySaml2
   public static class GenericSaml2Configuration {
   }
 
+  @Configuration
+  @ConditionalOnProperty(name = "cognizone.security.auth-method", havingValue = "disabled")
```
Comment (natan-cox-cognizone) [inline security/src/main/java/zone/cogni/lib/security/common/SecurityConfiguration.java]: should the value `disabled` and the `@EnableSecurityOff` not be aligned? well... that only makes sense if you have both exposed 🤔

File: security/src/main/java/zone/cogni/lib/security/common/SecurityConfiguration.java [hunk 2]
```diff
@@ -69,6 +76,10 @@ static class BasicAuthCondition {
     static class Saml2Condition {
     }
 
+    @ConditionalOnProperty(name = "cognizone.security.auth-method", havingValue = "disabled")
+    static class DisabledCondition {
```
Comment (natan-cox-cognizone) [inline security/src/main/java/zone/cogni/lib/security/common/SecurityConfiguration.java]: huh?

File: security/src/main/java/zone/cogni/lib/security/common/SecurityConfiguration.java (hunk 1 above)
Comment (domivds) [reply security/src/main/java/zone/cogni/lib/security/common/SecurityConfiguration.java]: yes, will rename the property to `off` (@EnableSecurityDisabled is too weird)

File: security/src/main/java/zone/cogni/lib/security/common/SecurityConfiguration.java (hunk 2 above)
Comment (domivds) [reply security/src/main/java/zone/cogni/lib/security/common/SecurityConfiguration.java]: kewl spring stuff 😛 
to support generic config with property and @EnableSecurityXXXX, I convert the property to a Configuration which goes back to @EnableSecurityXXXX
Then I use condition classes to find the good one or fail if value is wrong.

I will add a bit of comment about what happens

File: security/src/main/java/zone/cogni/lib/security/common/SecurityConfiguration.java (hunk 1 above)
Comment (natan-cox-cognizone) [reply security/src/main/java/zone/cogni/lib/security/common/SecurityConfiguration.java]: true

**Review by natan-cox-cognizone — APPROVED** [review]

**Pull request merged**